### PR TITLE
feat: implement subscriptions filter

### PR DESCRIPTION
**Pull request by jensneuse**

### CODE REVIEW

File: v2/pkg/engine/plan/configuration_visitor.go [hunk 1]
```diff
@@ -801,6 +804,101 @@ func (c *configurationVisitor) addNewPlanner(ref int, typeName, fieldName, curre
...
+			filter.Values[i].Segments = []resolve.TemplateSegment{
+				{
+					SegmentType: resolve.StaticSegmentType,
+					Data:        []byte(value),
+				},
+			}
+			continue
+		}
+		if len(matches) == 1 && len(matches[0]) == 4 {
+			prefix := value[:matches[0][0]]
```
Comment (devsergiy) [inline v2/pkg/engine/plan/configuration_visitor.go:870]: could you add some explanations about indexes? [0][2] not so clear why it is like this

Comment (jensneuse) [reply v2/pkg/engine/plan/configuration_visitor.go:870]: It's what the std Library gives back.
One match with 2 sub matches is a slice of one with a slice of 4, two indices marking start and end of the match.

Comment (devsergiy) [reply v2/pkg/engine/plan/configuration_visitor.go:870]: please add input examples which regexp should match

Comment (jensneuse) [reply v2/pkg/engine/plan/configuration_visitor.go:870]: done

**Pull request merged**